client/lib/feed-store/index.js

```diff
@@ -30,6 +30,11 @@ var feeds = {},
 function setFeed( feedId, feed ) {
 	if ( feed !== feeds[ feedId ] ) {
 		feeds[ feedId ] = feed;
+
+		if ( feedId !== feed.feed_ID ) {
```

Comment:
how was this called such that feed.feed_ID !== feedId? We should be normalizing at the call site, not in here.